api/src/test/java/org/apache/iceberg/transforms/TestDatesProjection.java

```diff
@@ -50,10 +55,17 @@ public void assertProjectionStrict(PartitionSpec spec, UnboundPredicate<?> filte
 
     Assert.assertEquals(expectedOp, predicate.op());
 
-    Literal literal = predicate.literal();
     Dates transform = (Dates) spec.getFieldsBySourceId(1).get(0).transform();
-    String output = transform.toHumanString((int) literal.value());
-    Assert.assertEquals(expectedLiteral, output);
+    if (predicate.op() == Expression.Operation.IN || predicate.op() == Expression.Operation.NOT_IN) {
```

Comment:
Strict projection never produces IN, so I think this should assert that the projection is never an IN predicate instead of handling it here.